src/components/Footer.tsx

```diff
@@ -0,0 +1,104 @@
+import classNames from 'classnames';
+import { Todo } from '../types/Todo';
+import { FilterMethods } from '../types/Methods';
+import { handleDelete } from '../functions';
+
+interface FooterProps {
+  checkCount: (todos: Todo[]) => number;
+  setFilter: (string: FilterMethods) => void;
+  todos: Todo[];
+  filterMethod: FilterMethods;
+  renderTodos: (todos: Todo[]) => void;
+  setDeleting: (todo: Todo) => void;
+  newError: (value: string) => void;
+}
+
+export const Footer: React.FC<FooterProps> = ({
+  checkCount,
+  setFilter,
+  todos,
+  filterMethod,
+  renderTodos,
+  setDeleting,
+  newError,
+}) => {
+  const clearCompleted = () => {
+    for (const todo of todos) {
+      if (todo.completed === true) {
+        handleDelete(todo, setDeleting, renderTodos, newError);
+      }
+    }
+  };
+
+  return (
+    <footer className="todoapp__footer" data-cy="Footer">
+      <span className="todo-count" data-cy="TodosCounter">
+        {`${checkCount(todos)} items left`}
+      </span>
+
+      {/* Active link should have the 'selected' class */}
+      <nav className="filter" data-cy="Filter">
+        <a
+          href="#/"
+          className={classNames(
+            filterMethod === 'All' ? 'filter__link selected' : 'filter__link',
+          )}
+          data-cy="FilterLinkAll"
+          onClick={e => {
+            e.preventDefault();
+            setFilter(FilterMethods.All);
+          }}
+        >
+          All
+        </a>
+
+        <a
+          href="#/active"
+          className={classNames(
+            filterMethod === 'Active'
+              ? 'filter__link selected'
+              : 'filter__link',
+          )}
+          data-cy="FilterLinkActive"
+          onClick={e => {
+            e.preventDefault();
+            setFilter(FilterMethods.Active);
+          }}
+        >
+          Active
+        </a>
+
+        <a
+          href="#/completed"
```

Comment:
Use `Object.values(your created enum)` and render these option with `map()` method